atoma-state/src/handlers.rs

```diff
@@ -782,21 +782,37 @@ pub(crate) async fn handle_state_manager_event(
         }
         AtomaAtomaStateManagerEvent::UpdateFiatAmount {
             user_address,
+            model_name,
             estimated_input_amount,
             input_amount,
+            input_tokens,
             estimated_output_amount,
             output_amount,
+            output_tokens,
         } => {
             state_manager
                 .state
                 .update_fiat_amount(
-                    user_address,
+                    user_address.clone(),
                     estimated_input_amount,
                     input_amount,
                     estimated_output_amount,
                     output_amount,
                 )
                 .await?;
+            if output_amount > 0 {
```

Comment:
Even the prefill phase generates at least one new token. When can it `0`?

Reply:
when something went wrong, there is a plenty of usage where we send 0